### DISCUSSION TITLE
How to add custom properties to requestBody?

### DISCUSSION THREAD

**Discussion by zorca** [dedoc/scramble#556] [Q&A] [answered]
I'm trying to add a required: true parameter for the entire requestBody object, but I don't understand how to do it using the extension. Maybe someone has an example?

**Comment by romalytvynenko** [comment]
@zorca I don't understand what exactly you want to achieve. Can you show the example of the expected OpenApi spec?

**Reply by zorca** [reply]
An example from my colleague's API specification
```
"requestBody": {
    "required": true,
    "content": {
        "application\/json": {
            "schema": {
                "type": "object",
                        "properties": {
                    "email": {
                        "type": "string",
                                "format": "email"
                            },
                            "password": {
                        "type": "string"
                            }
                        },
                        "required": [
                    "email",
                    "password"
                ]
            }
        }
    }
},
```

**Reply by zorca** [reply]
As you can see, requestBody contains the parameter `"required": true`

**Reply by zorca** [reply]
And working full example: https://app.swaggerhub.com/apis/STEELPUPPY_1/laravel/0.0.1#/Auth/auth.login

**Reply by manuel-watchenterprise** [reply]
@romalytvynenko +1

Being able to mark the full requestBody object as required is not only useful when documenting the API but code generators are using as well to mark a request call's param as optional or mandatory.

The current implementation of [RequestBodyObject](https://github.com/dedoc/scramble/blob/main/src/Support/Generator/RequestBodyObject.php) does not allow this even from custom extension.

```php
class RequestBodyObject
{
    public bool $required = false;

    public function required(bool $required = true): self
    {
        $this->required = $required;

        return $this;
    }

    public function toArray(): array
    {
        $result = array_filter([
            'description' => $this->description,
            'required' => $this->required,
        ]);

        // ...

        return $result;
    }
}
```

**Reply by manuel-watchenterprise** [reply, marked as answer]
Thank you @romalytvynenko for this feature! 🚀 
#580
